### PR TITLE
Update/SOF-7461 Update: various cleanups

### PR DESCRIPTION
**Pull request by VsevolodX**
update functions in modify to reflect the type of operand - material, interface - as the first or last word in the name [feature/SOF-7426](https://github.com/Exabyte-io/made/pull/158/files#diff-2cb3259519fb8e4e64b58639d7bead4ba488c0af31cd94a10cc3d7f7a46ed4eaR447)

Setup logging function(s) per https://github.com/Exabyte-io/made/pull/163/files#diff-e4a26d8b67b5aeea2b22353665f197977c0e02fc72c85eaf083509447df8bf11R316

create_xxx in every __init__ alongside imports of configs, builders and params

Functionality:

Use local max z detection in Adatom placement (to place on terrace or non-orthogonal surface)

Hetero material island

### CODE REVIEW

File: src/py/mat3ra/made/tools/analyze.py [hunk 1]
```diff
@@ -492,6 +492,8 @@ def get_local_extremum_atom_index(
     Returns:
         int: id of the atom with the minimum or maximum z-coordinate.
     """
+    if len(coordinate) == 2:
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/analyze.py:495]: This should be handled before invoking the function

File: src/py/mat3ra/made/tools/build/defect/builders.py [hunk 2]
```diff
@@ -174,7 +175,11 @@ def create_adatom(
     def _calculate_coordinate_from_position_and_distance(
         self, material: Material, position_on_surface: List[float], distance_z: float
     ) -> List[float]:
-        max_z = get_atomic_coordinates_extremum(material, use_cartesian_coordinates=False)
+        # max_z = get_atomic_coordinates_extremum(material, use_cartesian_coordinates=False)
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/build/defect/builders.py:178]: Remove

File: tests/py/unit/fixtures.py [hunk 3]
```diff
@@ -178,91 +178,139 @@
...
-    "name": "Si8(001), termination Si_P4/mmm_1, Slab",
     "isNonPeriodic": False,
     "_id": "",
     "metadata": {
         "boundaryConditions": {"type": "pbc", "offset": 0},
         "build": {
-            "configuration": SI_SLAB_CONFIGURATION,
-            "termination": "Si_P4/mmm_1",
+            "termination": "Si_P6/mmm_1",
+            "configuration": {
```
Comment (timurbazhirov) [inline tests/py/unit/fixtures.py:216]: Let's keep SI_SLAB_CONFIGURATION

File: tests/py/unit/test_tools_build_defect.py [hunk 4]
```diff
@@ -109,7 +107,9 @@ def test_create_crystal_site_adatom():
 
     assert defect.basis.elements.values[-1] == "Si"
     assertion_utils.assert_deep_almost_equal(
-        [0.083333333, 0.458333333, 0.352272727], defect.basis.coordinates.values[-1]
+        # TO pass on GHA
```
Comment (timurbazhirov) [inline tests/py/unit/test_tools_build_defect.py:110]: Explain what's happening here in a comment

File: tests/py/unit/test_tools_build_grain_boundary.py [hunk 5]
```diff
@@ -55,9 +55,10 @@ def test_slab_grain_boundary_builder():
     expected_lattice_vectors = [
         [25.140673461, 0.0, 0.0],
         [0.0, 3.867, 0.0],
-        [0.0, 0.0, 11.601],
+        [0.0, 0.0, 8.734],
     ]
-    expected_coordinate_15 = [0.777190818, 0.0, 0.083333333]
+    # To pass on GHA
```
Comment (timurbazhirov) [inline tests/py/unit/test_tools_build_grain_boundary.py:60]: Same as above

File: tests/py/unit/test_tools_build_slab.py [hunk 6]
```diff
@@ -12,11 +12,12 @@ def test_build_slab():
...
         thickness=1,
-        vacuum=1,
+        vacuum=5.0,
         xy_supercell_matrix=[[1, 0], [0, 1]],
         use_orthogonal_z=True,
         make_primitive=True,
     )
     termination = get_terminations(slab_config)[0]
     slab = create_slab(slab_config, termination)
+    print(slab.to_json())
```
Comment (timurbazhirov) [inline tests/py/unit/test_tools_build_slab.py:22]: Remove

File: src/py/mat3ra/made/tools/analyze.py (hunk 1 above)
Comment (VsevolodX) [reply src/py/mat3ra/made/tools/analyze.py:495]: I'll use decorator

File: src/py/mat3ra/made/tools/utils/__init__.py [hunk 7]
```diff
@@ -42,6 +42,35 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def decorator_convert_position_to_coordinate(func: Callable) -> Callable:
+    """
+    A decorator that converts a 2D position [x,y] to a 3D coordinate [x,y,0.0].
```
Comment (timurbazhirov) [inline src/py/mat3ra/made/tools/utils/__init__.py:47]: Spaces after `,`

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**